src/component/Search.js

```diff
@@ -0,0 +1,64 @@
+import React,{ useState , useEffect} from 'react';
+import axios from 'axios';
+
+const Search =()=>{
+    const [term,setTerm]=useState("");
+    const [results,setResults]=useState([]);
+    const resultRender=results.map((result)=>{
+        return (
+            <div key={result.pageid} className="item">
+                 <div className="right floated content">
+                    <a className="ui button" href={`https://en.wikipedia.org?curid=${result.pageid}`}>Go</a>
+                </div>
+                <div className="content">
+                    <div className="headers">
+                        {result.title}
+                    </div>
+                    <span dangerouslySetInnerHTML={{__html:result.snippet}} ></span>
+                </div>
+            </div>
+        )
+    });
+    useEffect( ()=>{
+        const search=async ()=>{
+            const {data}=await axios.get('https://en.wikipedia.org/w/api.php',{
+                params:{
+                    action:'query',
+                    list:'search',
+                    origin:'*',
+                    format:'json',
+                    srsearch:term,
+                }
+            });
+            setResults(data.query.search);
+        };
+        const timeoutid=setTimeout(() => {
```

Comment:
What is the need to use setTimeout?

Reply:
the purpose of using setTimeout  is to wait for time. In this we want to wait some amount of time if user stop pressing the key then make search